cmake with petsc only works once

`cmake -DBOUT_USE_PETSC=ON  ..` only works once.
The second time it fails with:
```
-- FFTW support: ON
-- LAPACK support: ON
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  PETSc could not be found.  Be sure to set PETSC_DIR and PETSC_ARCH.
  (missing: PETSC_EXECUTABLE_RUNS)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPETSc.cmake:376 (find_package_handle_standard_args)
  CMakeLists.txt:706 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/dave/soft/BOUT-dev/build/CMakeFiles/CMakeOutput.log".
See also "/home/dave/soft/BOUT-dev/build/CMakeFiles/CMakeError.log".
```
removing `CMakeCache.txt` is a workaround.

This is on fedora, I think `pkg-config` finds PETSc.

--- a/cmake/FindPETSc.cmake
+++ b/cmake/FindPETSc.cmake
@@ -344,19 +344,23 @@ int main(int argc,char *argv[]) {
   set (PETSC_INCLUDES ${petsc_includes_needed} CACHE STRING "PETSc include path" FORCE)
   set (PETSC_LIBRARIES ${PETSC_LIBRARIES_ALL} CACHE STRING "PETSc libraries" FORCE)
   set (PETSC_COMPILER ${petsc_cc} CACHE FILEPATH "PETSc compiler" FORCE)
-  # Note that we have forced values for all these choices.  If you
-  # change these, you are telling the system to trust you that they
-  # work.  It is likely that you will end up with a broken build.
-  mark_as_advanced (PETSC_INCLUDES PETSC_LIBRARIES PETSC_COMPILER PETSC_DEFINITIONS PETSC_MPIEXEC PETSC_EXECUTABLE_RUNS)
 endif ()
 
-if (NOT PETSC_INCLUDES)
-  include(FindPkgConfig)
-  pkg_search_module(PkgPETSC PETSc>3.4.0 petsc>3.4.0)
-  set (PETSC_LIBRARIES ${PkgPETSC_LINK_LIBRARIES} CACHE STRING "PETSc libraries" FORCE)
-  set (PETSC_INCLUDES ${PkgPETSC_INCLUDE_DIRS} CACHE STRING "PETSc include path" FORCE)
-  set (PETSC_EXECUTABLE_RUNS "not-needed")
+if (NOT PETSC_INCLUDES AND NOT TARGET PETSc::PETSc)
+  find_package(PkgConfig)
+  if (PkgConfig_FOUND)
+    pkg_search_module(PkgPETSC PETSc>3.4.0 petsc>3.4.0)
+    set (PETSC_LIBRARIES ${PkgPETSC_LINK_LIBRARIES} CACHE STRING "PETSc libraries" FORCE)
+    set (PETSC_INCLUDES ${PkgPETSC_INCLUDE_DIRS} CACHE STRING "PETSc include path" FORCE)
+    set (PETSC_EXECUTABLE_RUNS "YES" CACHE BOOL
+        "Can the system successfully run a PETSc executable?  This variable can be manually set to \"YES\" to force CMake to accept a given PETSc configuration, but this will almost always result in a broken build.  If you change PETSC_DIR, PETSC_ARCH, or PETSC_CURRENT you would have to reset this variable." FORCE)
+  endif()
 endif()
+
+# Note that we have forced values for all these choices.  If you
+# change these, you are telling the system to trust you that they
+# work.  It is likely that you will end up with a broken build.
+mark_as_advanced (PETSC_INCLUDES PETSC_LIBRARIES PETSC_COMPILER PETSC_DEFINITIONS PETSC_MPIEXEC PETSC_EXECUTABLE_RUNS)
 
 include (FindPackageHandleStandardArgs)
 find_package_handle_standard_args (PETSc

